test/epocxy/cxy_ctl_SUITE.erl

```diff
@@ -515,3 +517,36 @@ check_copying_dict(_Config) ->
 
     ok.
 
+check_high_water(_Config) ->
+    {Inline_Type, Spawn_Type} = {high_water_inline, high_water_spawn},
+    Spawn_Max = 2,
+    Limits = [{Inline_Type, inline_only, 0, ?MAX_SLOW_FACTOR}, {Spawn_Type, 2, 0, ?MAX_SLOW_FACTOR}],
+    true = ?TM:init(Limits),
+
+    %% Inline task, never run more than one at a time.
+    ?TM:high_water(Inline_Type, clear),
+    0 = ?TM:high_water(Inline_Type),
+    ok = ?TM:execute_task(Inline_Type, timer, sleep, [200]), erlang:yield(),
+    ok = ?TM:execute_task(Inline_Type, timer, sleep, [200]), erlang:yield(),
+    1 = ?TM:high_water(Inline_Type),
+    1 = ?TM:high_water(Inline_Type, no_clear),
+    1 = ?TM:high_water(Inline_Type, clear),
+    0 = ?TM:high_water(Inline_Type),
+
+    %% Spawn task, run concurrently.
+    ?TM:high_water(Spawn_Type, clear),
+    0 = ?TM:high_water(Spawn_Type),
+    ok = ?TM:execute_task(Spawn_Type, timer, sleep, [200]), erlang:yield(),
+    ok = ?TM:execute_task(Spawn_Type, timer, sleep, [200]), erlang:yield(),
+    2 = ?TM:high_water(Spawn_Type, clear),
+
+    %% Spawn task, run concurrently but not more than max.
+    ok = ?TM:execute_task(Spawn_Type, timer, sleep, [200]), erlang:yield(),
+    ok = ?TM:execute_task(Spawn_Type, timer, sleep, [200]), erlang:yield(),
+    ok = ?TM:execute_task(Spawn_Type, timer, sleep, [200]), erlang:yield(),
+    ok = ?TM:execute_task(Spawn_Type, timer, sleep, [200]), erlang:yield(),
+    ok = ?TM:execute_task(Spawn_Type, timer, sleep, [200]), erlang:yield(),
+    %% 3 because 2 are spawned and 1 runs inline concurrently
+    3 = ?TM:high_water(Spawn_Type),
+
+    ok.
```

Comment:
Nice demonstration and example code. The 3 is unexpected but important to realize. Thanks for the good test! I will eventually add PropEr testing.